scheduler/main.go

```diff
@@ -35,41 +34,30 @@ func main() {
 	if err != nil {
 		panic(err)
 	}
-
-	for {
-		msg, err := sub.Receive(ctx)
-		if err != nil {
-			log.Println("error receiving message: ", err)
-			continue
+	srv := proxy.New(topic, sub)
+	fmt.Println("Listening for messages to proxy...")
+
+	err = srv.Listen(ctx, func(m *pubsub.Message) (*pubsub.Message, error) {
+		log.Println("handling message: ", string(m.Body))
+		pkg := feeds.Package{}
+		if err := json.Unmarshal(m.Body, &pkg); err != nil {
+			return nil, fmt.Errorf("error unmarshalling json: %w", err)
 		}
-		go func(m *pubsub.Message) {
-			log.Println("handling message: ", string(m.Body))
-			pkg := library.Package{}
-			if err := json.Unmarshal(m.Body, &pkg); err != nil {
-				log.Println("error unmarshalling json: ", err)
-				return
-			}
-			if err := handlePkg(ctx, topic, pkg); err != nil {
-				fmt.Println("Error: ", err)
-				msg.Nack()
-				return
-			}
-			msg.Ack()
-		}(msg)
-	}
-}
+		if _, ok := supportedPkgManagers[pkg.Type]; !ok {
+			err = fmt.Errorf("package type is not supported: %v", pkg.Type)
+			return nil, err
```

Comment:
nit: no need for the assignment to `err`, just `fmt.Errorf` directly here as on line 44.